### ISSUE TITLE
Jekyll issue #6163 introduced a breaking change.

### ISSUE THREAD

**Issue by envygeeks** [jekyll/jekyll#6249]
#6163 introduced a breaking change that can affect anybody who uses plugins. Between Jekyll 3.5.0 and Jekyll 3.5.1. The commit that does this can be found here https://github.com/jekyll/jekyll/commit/7a853762467d410ca84e7570758ec19bcd0cb66f

In Jekyll 3.5.0:

```
[2] pry(#<#<Class:0x00557663ce3590>>)> show-method site

From: /srv/jekyll/vendor/bundle/ruby/2.3.0/gems/jekyll-3.5.0/lib/jekyll/filters/url_filters.rb @ line 43:
Owner: Jekyll::Filters::URLFilters
Visibility: private
Number of lines: 3

def site
  @context.registers[:site]
end
```

In Jekyll 3.5.1 if anybody relies on said method, it automatically breaks their source, without warning, without reason, between a patch/bugfix version.  Even though this method was "private" it was available because of the way that Liquid handles it's filters, and the fact that Jekyll chose to make it private instead of protected.

/cc @ashmaroli

**Comment by ashmaroli** [comment]
I chose to remove that method because it unintentionally broke many in-house plugins (additional ref: [comment](https://github.com/jekyll/jekyll/issues/6154#issuecomment-309427209))

v3.5.1 is majorly about reverting that one change albeit released as a patch version. I personally do not think it would be a good idea to re-introduce the private method.
**Instead we should encourage downstream usage to be altered according to the current Jekyll state.**

I agree its wrong when a patch release breaks something. But, the onus is also on those who chose to rely on a buggy method.

/cc @jekyll/ecosystem @benbalter

**Comment by envygeeks** [comment]
I personally rely on that method and I don't feel that i should need to change in a minor version, it's silly to even expect me to; this software adheres to the SemVer spec, which means this should have never happened in 3.5.1 to begin with regardless of any opinions on it being buggy.

Furthermore, I don't understand how a single line method that does one thing, does it well, and only access a very consistent and readily available `@context` and `register` can be buggy.  You are implying that `@context` is buggy.  Which is a Liquid problem at that point. It wasn't the method that was buggy, it was a bad implementation which is something that Jekyll should have to foot the bill for until the next major version as it's breaking the contract it created with Plugin authors.  I also feel that the implementation details listed in that comment seem to be mixing concerns in a dangerous and buggy way resulting in us having to foot the bill on a contract change we didn't agree to.

**Comment by envygeeks** [comment]
What I mean is that if you inherit from `Page` and then decide to bring in `URLFilters`, you are expecting one concern to change for another concern, this isn't how software should work and this isn't how coherent software design works either.

**Comment by pathawks** [comment]
It was a private method.
It was not part of any documented/supported API for plugins.
It wasn't available before 3.5.0.
It caused problems for users.
It was removed very quickly after 3.5.0 was released.
